msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -90,8 +101,50 @@ public Integer getNumberOfDataPoints() {
 
 	@Override
 	public double[] getMzValues() {
```

Comment:
As I said before, you should save the position and size of the encoded block. Then you can just read the number of bytes that is necessary.